### PR TITLE
compression

### LINKED ISSUE #47: Compression of messages
Hey,

Turns out brod doesnt support gzip/snappy which are part of the kafka wire protocol. I know you are heading towards 2.0. Will this be a feature in there? Do you want the code I wrote to support this? 

I am looking forward to 2.0 and the high level consumer, I ended up writing quite a bit of code to do a simple consumer with the current version.

### PR DESCRIPTION
**Pull request by zmstone**
fixes #47

### CODE REVIEW

**Comment by zmstone** [comment]
work in progress.
will add tests for snappy when kafka_protocol is ready.

**Comment by zmstone** [comment]
travis is failing most likely because git version was too old there causing the kafka_protocol dependency was not checkout at the right commit.

File: Makefile [hunk 1]
```diff
@@ -4,7 +4,7 @@ PROJECT_VERSION = 2.0-dev
 
 DEPS = supervisor3 kafka_protocol
 
-dep_kafka_protocol_commit = 0.2.3
+dep_kafka_protocol = git https://github.com/klarna/kafka_protocol.git add-compression
```
Comment (id) [inline Makefile]: Update before merging

**Comment by id** [comment]
Looks good. See comment above + update TODO = good to merge.

**Pull request merged**